src/owner.rs

```diff
@@ -0,0 +1,108 @@
+use std::{mem, slice};
+
+use ndarray::{ArrayBase, Dimension, OwnedRepr};
+use pyo3::class::impl_::{PyClassImpl, ThreadCheckerStub};
+use pyo3::pyclass::PyClass;
+use pyo3::pyclass_slots::PyClassDummySlot;
+use pyo3::type_object::{LazyStaticType, PyTypeInfo};
+use pyo3::{ffi, types::PyAny, PyCell};
+
+use crate::dtype::Element;
+
+pub(crate) struct Owner {
+    ptr: *mut u8,
+    len: usize,
+    cap: usize,
+    drop: unsafe fn(*mut u8, usize, usize),
+}
```

Comment:
Isn't `Owner` a too broad name? I prefer to `PySliceContainer`, `SliceContainer`, `SliceOwner` and so on.

Reply:
Went back to the "container" naming, i.e. `PySliceContainer` for the type and `container: C` for the bindings.